src/wp-includes/media.php

```diff
@@ -793,6 +803,356 @@ function wp_get_attachment_image($attachment_id, $size = 'thumbnail', $icon = fa
+ * the purpose of populating a source set when creating responsive image markup.
+ *
+ * @since 4.4.0
+ *
+ * @param  int    $id   Image attachment ID.
+ * @param  string $size Optional. Name of image size. Default value: 'medium'.
+ * @return array|bool  An array of image urls and widths or false on failure.
+ */
+function wp_attachment_img_srcset_array( $id, $size = 'medium' ) {
+	// Get the intermediate size.
+	$image = image_get_intermediate_size( $id, $size );
+	// Get the post meta.
+	$img_meta = wp_get_attachment_metadata( $id );
+
+	// Extract the height and width from the intermediate or the full size.
+	$img_width = ( $image ) ? $image['width'] : $img_meta['width'];
+	$img_height = ( $image ) ? $image['height'] : $img_meta['height'];
+
+	// Bail early if the width isn't greater that zero.
+	if ( ! $img_width > 0 ) {
+		return false;
+	}
+
+	// Use the url from the intermediate size or build the url from the metadata.
+	if ( ! empty( $image['url'] ) ) {
+		$img_url = $image['url'];
+	} else {
+		$uploads_dir = wp_upload_dir();
+		$img_file = ( $image ) ? path_join( dirname( $img_meta['file'] ) , $image['file'] ) : $img_meta['file'];
+		$img_url = $uploads_dir['baseurl'] . '/' . $img_file;
+	}
+
+	$img_sizes = $img_meta['sizes'];
+
+	// Add full size to the img_sizes array.
+	$img_sizes['full'] = array(
+		'width'  => $img_meta['width'],
+		'height' => $img_meta['height'],
+		'file'   => wp_basename( $img_meta['file'] )
+	);
+
+	// Calculate the image aspect ratio.
+	$img_ratio = $img_height / $img_width;
+
+	/*
+	 * Images that have been edited in WordPress after being uploaded will
+	 * contain a unique hash. Look for that hash and use it later to filter
+	 * out images that are leftovers from previous versions.
+	 */
+	$img_edited = preg_match( '/-e[0-9]{13}/', $img_url, $img_edit_hash );
+
+	// Set up an array to hold the matched image sources.
+	$sources = array();
+
+	/*
+	 * Loop through available images and only use images that are resized
+	 * versions of the same rendition.
+	 */
+	foreach ( $img_sizes as $img ) {
+
+		// Filter out images that are leftovers from previous renditions.
+		if ( $img_edited && ! strpos( $img['file'], $img_edit_hash[0] ) ) {
+			continue;
+		}
+
+		// Calculate the new image ratio.
+		$img_ratio_compare = $img['height'] / $img['width'];
+
+		// If the new ratio differs by less than 0.01, use it.
+		if ( abs( $img_ratio - $img_ratio_compare ) < 0.01 ) {
+			$sources[ $img['width'] ] = path_join( dirname( $img_url ), $img['file'] ) . ' ' . $img['width'] .'w';
+		}
+	}
+
+	/**
+	 * Filter the output of wp_get_srcset_array().
+	 *
+	 * @since 4.4.0
+	 *
+	 * @param array        $sources  An array of image urls and widths.
+	 * @param int          $id       Attachment ID for image.
+	 * @param array|string $size     Size of image, either array or string.
+	 */
+	return apply_filters( 'wp_attachment_img_srcset', $sources, $id, $size );
```

Comment:
I believe this was a back compatibility issue when we separated the `tevkori_get_srcset_array` function from `tevkori_get_srcset`. We can rename this filter to match the name of the wrapping function.